libs/antsi/src/config.rs

```diff
@@ -0,0 +1,130 @@
+//! # Configuration
+//!
+//! There are two predominant "industry" standards when it comes to parameters, parameters are used
+//! where one needs to specify more than a single instruction, this is predominately used for
+//! advanced colors, like [`crate::IndexedColor`], or [`crate::RgbColor`].
+//!
+//! The two differences across implementations are:
+//! * [`Delimiter`]
+//! * [`Compliance`]
+//!
+//! For more information please look at the respective documentation. `antsi` uses
+//! [`Delimiter::Colon`] and [`Compliance::Iso`] as default in all documentation.
+// TODO: recommendations for most used terminal emulators
+use core::sync::atomic::{AtomicU8, Ordering};
```

Comment:
I'm not particularly happy to require those types. Why do we need to store the configuration globally? It implies, that we assume `antsi` is used in multithreaded applications, but IMO `antsi` should not be agnostic over those assumptions. Why don't we let the downstream crates handle this behavior?

Reply:
> a user outputs their content into a single type of terminal

I think this assumption is wrong, e.g. when writing to a terminal _and_ to a log file you may want different formats.

Reply:
🤔 even then, we need a way to globally configure this to stay consistent while allowing for compatibility with older terminals and I feel like consistency across all formatting should be a top priority, as to not confuse the library user or end user of the application. This is also what all current applications do, they use a specific combination (usually `;` + `Legacy`) and then stick w/ it, but I'd like to give the user the choice of what that combination is. 🤔 

We *could*, of course, also do something similar to: `let _guard = Delimiter::Color.set()`. If we wanted to output to both, the log file stream could use that instead when interpolating the `Display`.

The log file is a great example, as we have another problem: if we want to output to a log file, meaning non-interactive, one should be able to strip color. This is what all other apps usually do. This, of course, could also be done by a downstream crate.

Reply:
We have decided offline to remove the configuration part. To properly support nested color reset (e.g. `[blue]Value [red]Value[/red] Value[/blue]`), we will need a stateful printer anyway. Configuration of the delimiter will default to the legacy version and be configurable on the printer itself.